integrations/octopus/main.py

```diff
@@ -48,25 +48,57 @@ async def setup_application() -> None:
         )
         return
     octopus_client = await init_client()
-    existing_subscriptions = await octopus_client.get_webhook_subscriptions()
-    existing_webhook_uris = {
-        subscription.get("EventNotificationSubscription", {}).get("WebhookURI")
-        for subscription in existing_subscriptions
-    }
-    webhook_uri = f"{app_host}/integration/webhook"
-    if webhook_uri in existing_webhook_uris:
-        logger.info(f"Webhook already exists with URI: {webhook_uri}")
-    else:
-        await octopus_client.create_webhook_subscription(app_host)
-        logger.info(f"Webhook created with URI: {webhook_uri}")
+    async for spaces in octopus_client.get_all_spaces():
+        for space in spaces:
+            async for subscriptions in octopus_client.get_webhook_subscriptions(
+                space.get("Id")
+            ):
+                existing_webhook_uris = {
+                    subscription.get("EventNotificationSubscription", {}).get(
+                        "WebhookURI"
+                    )
+                    for subscription in subscriptions
+                }
+                existing_webhook_names = [
+                    subscription.get("Name", "") for subscription in subscriptions
+                ]
+                webhook_uri = f"{app_host}/integration/webhook"
+                if webhook_uri in existing_webhook_uris:
+                    logger.info(f"Webhook already exists with URI: {webhook_uri}")
+                elif any(
+                    name.startswith("Port Subscription -")
+                    for name in existing_webhook_names
+                ):
+                    logger.info(
+                        "Webhook already exists with name starting with 'Port Subscription -'"
+                    )
+                else:
+                    await octopus_client.create_webhook_subscription(
+                        app_host, space.get("Id")
+                    )
+                    logger.info(f"Webhook created with URI: {webhook_uri}")
 
 
 @ocean.on_resync()
 async def resync_resources(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    if kind == ObjectKind.SPACE:
+        return
     octopus_client = await init_client()
-    async for resource_batch in octopus_client.get_paginated_resources(kind):
-        logger.info(f"Received length  {len(resource_batch)} of {kind} ")
-        yield resource_batch
+    async for spaces in octopus_client.get_all_spaces():
+        for space in spaces:
+            async for resource_batch in octopus_client.get_paginated_resources(
+                kind, path_parameter=space.get("Id")
+            ):
+                logger.info(f"Received batch of {len(resource_batch)} {kind} ")
+                yield resource_batch
+
```

Comment:
Since we have nested pagination, we can use this helper function

from port_ocean.utils.async_iterators import stream_async_iterators_tasks


                tasks = [
                    get paginated resource (kind, space.get("Id"))
                    for space in spaces
                ]
                resource_data = []
                async for batch in stream_async_iterators_tasks(*tasks):
                    yield batch

Reply:
Done, thank you for the insight